segmentstore/storage/src/main/java/io/pravega/segmentstore/storage/chunklayer/ReadOperation.java

```diff
@@ -292,13 +329,26 @@ private void logEnd() {
                         SLTS_READ_INDEX_SCAN_LATENCY.reportSuccessEvent(elapsed);
                         SLTS_READ_INDEX_NUM_SCANNED.reportSuccessValue(cntScanned.get());
                     }
+
+                    // Prefetch possible chunks for next read.
+                    if (chunkToReadFrom.getNextChunk() != null) {
+                        // Do not wait.
+                        txn.get(chunkToReadFrom.getNextChunk());
+                    }
+
                     log.debug("{} read - chunk lookup - op={}, segment={}, offset={}, scanned={}, latency={}.",
                             chunkedSegmentStorage.getLogPrefix(), System.identityHashCode(this),
                             handle.getSegmentName(), offset, cntScanned.get(), elapsed.toMillis());
                 }, chunkedSegmentStorage.getExecutor()),
         chunkedSegmentStorage.getExecutor());
     }
 
+    private long getFloorBlockStartOffset(long offsetToRead) {
+        val floorBlock = offsetToRead / chunkedSegmentStorage.getConfig().getIndexBlockSize();
+        val floorBlockStartOffset = floorBlock * chunkedSegmentStorage.getConfig().getIndexBlockSize();
+        return floorBlockStartOffset;
```

Comment:
`return offsetToRead - offsetToRead % chunkedSegmentStorage.getConfig().getIndexBlockSize()`

This should avoid the need for a division and then multiplication.

Reply:
I don't remember exact details, but I had an off by one bug earlier with something similar.
So I am not yet sure.
May be we keep it as it is for now.